linetools/spectra/xspectrum1d.py

```diff
@@ -1301,15 +1304,17 @@ def fit_continuum(self, knots=None, edges=None, wlim=None, dw=10.,
         import matplotlib.pyplot as plt
         fig = plt.figure(figsize=(11, 7))
         fig.subplots_adjust(left=0.05, right=0.95, bottom=0.1, top=0.95)
-        wrapper = InteractiveCoFit(wa, self.flux.value, self.sig.value,
+        wrapper = InteractiveCoFit(wa, flux, sig,
                                    contpoints, co=co_init, fig=fig, anchor=anchor)
 
         # wait until the interactive fitting has finished
         while not wrapper.finished:
             plt.waitforbuttonpress()
 
         print('Updating continuum')
-        self.data['co'] = wrapper.continuum
+        # import pdb; pdb.set_trace()
+        gdp = ~self.data['wave'][self.select].mask
```

Comment:
Not so sure anymore. Will wait until tomorrow morning to give it a second pass before merging then (and to solve the Python 3 error).